core-bundle/src/Csrf/MemoryTokenStorage.php

```diff
@@ -67,12 +67,19 @@ public function hasToken($tokenId): bool
     /**
      * {@inheritdoc}
      */
-    public function removeToken($tokenId): void
+    public function removeToken($tokenId): ?string
     {
         $this->assertInitialized();
 
+        $token = null;
         $this->usedTokens[$tokenId] = true;
-        $this->tokens[$tokenId] = null;
+
+        if (array_key_exists($tokenId, $this->tokens)) {
```

Comment:
Why did you add this?

Reply:
@aschempp @ausi According to the [interface](https://github.com/symfony/security/blob/master/Csrf/TokenStorage/TokenStorageInterface.php#L45), we are supposed to return `string|null`. However, we are not returning anything at the moment.

1. How can this have worked?
2. Are the changes correct in your opinion?

Reply:
> 1. How can this have worked?

Returning nothing is the same as `null` so there was no type error. We don’t use the return value of this method in our code so it probably didn’t have an effect for our use case.

> 2. Are the changes correct in your opinion?

Yes, not returing the token was a mistake :(  
We should probably test the return value in the unit test: https://github.com/contao/contao/blob/b71079a3b55d55950d87cd972e1d0f865c121d89/core-bundle/tests/Csrf/MemoryTokenStorageTest.php#L36-L41

Reply:
Completely agree with @ausi

Reply:
I have backported the fixes in 66803d67f0b3069470107438f86d2c8171ec132d.